art/wrappers/output_add_random_noise.py

```diff
@@ -0,0 +1,136 @@
+# MIT License
+#
+# Copyright (C) IBM Corporation 2019
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements wrapper adding random Gaussian noise to the classifier output.
+"""
+import logging
+
+import numpy as np
+
+from art.wrappers.wrapper import ClassifierWrapper
+from art.classifiers.classifier import Classifier
+
+logger = logging.getLogger(__name__)
+
+
+class RandomNoise(ClassifierWrapper, Classifier):
+    """
+    Implementation of a classifier wrapper adding random Gaussian noise to the classifier output.
+    """
+
+    def __init__(self, classifier, scale=0.2):
+        """
+        Create a wrapper for rounded prediction output.
+
+        :param classifier: A trained classifier.
+        :type classifier: :class:`.Classifier`
+        :param scale: Standard deviation of the distribution.
+        :type scale: `float`
+        """
+        super(RandomNoise, self).__init__(classifier)
+        self.scale = scale
+
+    # pylint: disable=W0221
+    def predict(self, x, batch_size=128, **kwargs):
+        """
+        Make a prediction with the wrapped classifier, add random Gaussian noise and re-normalize predicted output.
+
+        :param x: Input data
+        :type x: `np.ndarray`
+        :param batch_size: Batch size
+        :type batch_size: `int`
+        :return: Rounded predictions of shape `(nb_inputs, nb_classes)`.
+        :rtype: `np.ndarray`
+        """
+        predictions = self.classifier.predict(x, batch_size=batch_size)
```

Comment:
I just wonder if we would want to pass kwargs for future change (although it's not used by classifier.predict).

Reply:
Good idea, I'll add kwargs.